pkg/apis/experimental/types.go

```diff
@@ -461,3 +463,61 @@ type JobCondition struct {
 	// Human readable message indicating details about last transition.
 	Message string `json:"message,omitempty"`
 }
+
+// DerivedNodeMetricsList contains derived metrics for a single node. It holds
+// derived metrics in windows on the node itself and all of its system
+// containers.
+type DerivedNodeMetrics struct {
+	api.TypeMeta `json:",inline"`
+	// Standard list metadata.
+	api.ObjectMeta `json:"metadata,omitempty"`
+
+	// NodeMetrics contains derived metrics on the node.
+	NodeMetrics MetricsWindows `json:"nodeMetrics"`
+	// SystemContainers contains derived container metrics for all system
+	// containers in the node.
+	SystemContainers []DerivedContainerMetrics `json:"systemContainers"`
+}
+
+// DerivedNodeMetricsList contains derived metrics for a single container.
+type DerivedContainerMetrics struct {
+	// ContainerName is the name of the container.
+	ContainerName string `json:"containerName"`
+	// Windows contains derived metrics on the container.
+	Windows MetricsWindows `json:"windows"`
+}
+
+// DerivedNodeMetricsList contains derived node metrics for all nodes.
+type DerivedNodeMetricsList struct {
+	api.TypeMeta `json:",inline"`
+	// Standard list metadata.
+	api.ListMeta `json:"metadata,omitempty"`
+
+	// Items holds a list of a DerivedNodeMetrics object for every node.
+	Items []DerivedNodeMetrics `json:"items"`
+}
+
+// MetricsWindows holds multiple derived metrics windows. All the windows are
+// rolling.
+type MetricsWindows struct {
+	// EndTime is the end time of all the time windows.
+	EndTime util.Time `json:"endTime"`
+
+	// ByDuration maps the durations of windows to their derived metrics
+	// The durations are 1m, 1h and 24h (time.Duration doesn't do days).
+	ByDuration map[time.Duration]DerivedMetrics `json:"byDuration"`
+}
+
+// DerivedMetrics holds derived metrics for multiple resources.
+type DerivedMetrics struct {
+	// Mean holds the averages over the window.
+	Mean ResourceUsage `json:"mean"`
+	// Max holds the maximum values over the window.
+	Max ResourceUsage `json:"max"`
+	// NinetyFive holds the 95th percentile values over the window.
+	NinetyFive ResourceUsage `json:"95th"`
```

Comment:
The json field name should match the Go field name (modulo capitalization), and please don't start with a leading integer.

https://github.com/kubernetes/kubernetes/blob/master/docs/devel/api-conventions.md#naming-conventions